webpack.config.js

```diff
@@ -142,6 +144,7 @@ const plugins = [
   new webpack.EnvironmentPlugin({
     ENV: ENV,
     NODE_ENV: NODE_ENV === "production" ? "production" : "development",
+    SELF_HOSTED: SELF_HOSTED,
```

Comment:
This is a bit messy, do we need this?

I believe @joseph-flinn and I settled on using `NODE_ENV` to switch between development and production code parts, and `ENV` to switch between cloud, qa, self-hosted.

Reply:
we settled on `NODE_ENV` signifying isDev and `ENV` indicating the type of environment. This is a fourth environment.

We either need `ENV=development` and `ENV=self_hosted_development` to launch dev servers OR we need another env variable to signify that we're developing for self hosted. I like this way better than the three-way ENV switch

Reply:
It would have run as if it were cloud production, but it's all torn out now.